gtk4/src/subclass/drawing_area.rs

```diff
@@ -0,0 +1,60 @@
+use gtk_sys;
+
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::Cast;
+
+use super::widget::WidgetImpl;
+use DrawingArea;
+use Widget;
+
+pub trait DrawingAreaImpl: DrawingAreaImplExt + WidgetImpl {
+    fn resize(&self, drawing_area: &Self::Type, width: i32, height: i32) {
```

Comment:
Not sure why it has a vfunc for resizing but nothing for drawing... oh well! :)

Reply:
The drawing happens on snapshot at widget level iirc.

Reply:
& the resize was added as people always misused size-allocate for drawing area, thankfully that was removed